### ISSUE TITLE
No enum constant org.keycloak.models.IdentityProviderSyncMode.import when using google idp

### ISSUE THREAD

**Issue by jderuere** [keycloak/keycloak#12740]
### Describe the bug

I'm able to register with the google idp, but after I logged out and try to log back in, I got an error.

### Version

18.0.2

### Expected behavior

Google login working

### Actual behavior

```
2022-06-25 11:45:37,296 WARN  [org.keycloak.events] (executor-thread-100) type=IDENTITY_PROVIDER_LOGIN_ERROR, realmId=myrealm, clientId=app-ui, userId=null, ipAddress=172.18.0.1, error=identity_provider_login_failure, identity_provider=google, redirect_uri=http://localhost:4200/, identity_provider_identity=deruere.julien@gmail.com, code_id=749c1f5e-6fdd-4b44-9b0f-da935790ed12, authSessionParentId=749c1f5e-6fdd-4b44-9b0f-da935790ed12, authSessionTabId=gf6q4eN3E2s
2022-06-25T11:50:43.205711863Z 2022-06-25 11:50:43,204 ERROR [org.keycloak.broker.oidc.AbstractOAuth2IdentityProvider] (executor-thread-109) Failed to make identity provider oauth callback: java.lang.IllegalArgumentException: No enum constant org.keycloak.models.IdentityProviderSyncMode.import
2022-06-25T11:50:43.205802939Z 	at java.base/java.lang.Enum.valueOf(Enum.java:240)
2022-06-25T11:50:43.205816971Z 	at org.keycloak.models.IdentityProviderSyncMode.valueOf(IdentityProviderSyncMode.java:3)
2022-06-25T11:50:43.205822270Z 	at org.keycloak.models.IdentityProviderModel.getSyncMode(IdentityProviderModel.java:218)
2022-06-25T11:50:43.205839517Z 	at org.keycloak.services.resources.IdentityBrokerService.updateFederatedIdentity(IdentityBrokerService.java:973)
2022-06-25T11:50:43.205844354Z 	at org.keycloak.services.resources.IdentityBrokerService.authenticated(IdentityBrokerService.java:603)
2022-06-25T11:50:43.205848529Z 	at org.keycloak.broker.oidc.AbstractOAuth2IdentityProvider$Endpoint.authResponse(AbstractOAuth2IdentityProvider.java:514)
2022-06-25T11:50:43.205852684Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-06-25T11:50:43.205856710Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
2022-06-25T11:50:43.205860753Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-06-25T11:50:43.205864911Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
2022-06-25T11:50:43.205868917Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
2022-06-25T11:50:43.205873733Z 	at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
2022-06-25T11:50:43.205878110Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:660)
2022-06-25T11:50:43.205882299Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:524)
2022-06-25T11:50:43.205886415Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:474)
2022-06-25T11:50:43.205890454Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
2022-06-25T11:50:43.205894552Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:476)
2022-06-25T11:50:43.205898659Z 	at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:434)
2022-06-25T11:50:43.205903744Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:192)
2022-06-25T11:50:43.205908048Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:152)
2022-06-25T11:50:43.205912108Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invokeOnTargetObject(ResourceLocatorInvoker.java:183)
2022-06-25T11:50:43.205916188Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:141)
2022-06-25T11:50:43.205920197Z 	at org.jboss.resteasy.core.ResourceLocatorInvoker.invoke(ResourceLocatorInvoker.java:32)
2022-06-25T11:50:43.205924236Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
2022-06-25T11:50:43.205928244Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
2022-06-25T11:50:43.205932295Z 	at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
2022-06-25T11:50:43.205941811Z 	at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
2022-06-25T11:50:43.205946317Z 	at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
2022-06-25T11:50:43.205950401Z 	at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
2022-06-25T11:50:43.205954473Z 	at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
2022-06-25T11:50:43.205958596Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:151)
2022-06-25T11:50:43.205962720Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:82)
2022-06-25T11:50:43.205966861Z 	at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.handle(VertxRequestHandler.java:42)
2022-06-25T11:50:43.205971084Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
2022-06-25T11:50:43.205976607Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
2022-06-25T11:50:43.205980866Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
2022-06-25T11:50:43.205984939Z 	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:67)
2022-06-25T11:50:43.205989101Z 	at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:55)
2022-06-25T11:50:43.205993229Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
2022-06-25T11:50:43.205997238Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
2022-06-25T11:50:43.206001720Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
2022-06-25T11:50:43.206006064Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:380)
2022-06-25T11:50:43.206010113Z 	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$5.handle(VertxHttpRecorder.java:358)
2022-06-25T11:50:43.206014119Z 	at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1212)
2022-06-25T11:50:43.206018183Z 	at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:163)
2022-06-25T11:50:43.206022246Z 	at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
2022-06-25T11:50:43.206026349Z 	at org.keycloak.quarkus.runtime.integration.web.QuarkusRequestFilter.lambda$createBlockingHandler$1(QuarkusRequestFilter.java:71)
2022-06-25T11:50:43.206030550Z 	at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:159)
2022-06-25T11:50:43.206034610Z 	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:100)
2022-06-25T11:50:43.206038629Z 	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$1(ContextImpl.java:157)
2022-06-25T11:50:43.206042618Z 	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:543)
2022-06-25T11:50:43.206051404Z 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
2022-06-25T11:50:43.206055813Z 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
2022-06-25T11:50:43.206059915Z 	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
2022-06-25T11:50:43.206064040Z 	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
2022-06-25T11:50:43.206068236Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2022-06-25T11:50:43.206072357Z 	at java.base/java.lang.Thread.run(Thread.java:829)
2022-06-25T11:50:43.206076394Z 
2022-06-25T11:50:43.218315656Z 2022-06-25 11:50:43,217 WARN  [org.keycloak.events] (executor-thread-109) type=IDENTITY_PROVIDER_LOGIN_ERROR, realmId=myrealm, clientId=app-ui, userId=null, ipAddress=172.18.0.1, error=identity_provider_login_failure, identity_provider=google, redirect_uri=http://localhost:4200/, identity_provider_identity=deruere.julien@gmail.com, code_id=749c1f5e-6fdd-4b44-9b0f-da935790ed12, authSessionParentId=749c1f5e-6fdd-4b44-9b0f-da935790ed12, authSessionTabId=q5pC-hUIa0U
```

### How to Reproduce?

I'm using keycloak in Docker with image: quay.io/keycloak/keycloak:18.0.2

**Comment by rmartinc** [comment]
@jderuere It seems to me that the provider configuration is having some wrong value for the `syncMode` attribute. What do you see in the console for the field `Sync Mode`? (It's a combo with three values, import, legacy and force). 

Using kcadm in my provider used for testing I have something like this:

```
./kcadm.sh get identity-provider/instances/google -r test
{
  "alias" : "google",
  "internalId" : "e59526bb-3d67-432a-ae8f-5b399d747dd4",
  "providerId" : "google",
  "enabled" : true,
  "updateProfileFirstLoginMode" : "on",
  "trustEmail" : false,
  "storeToken" : false,
  "addReadTokenRoleOnCreate" : false,
  "authenticateByDefault" : false,
  "linkOnly" : false,
  "firstBrokerLoginFlowAlias" : "first broker login",
  "config" : {
    "syncMode" : "IMPORT",
    "clientSecret" : "**********",
    "clientId" : "XXXXXX",
    "useJwksUrl" : "true"
  }
}
```
As you see the `syncMode` is valid value IMPORT. What do you have in that field?

**Comment by jderuere** [comment]
Using the rest API (http://localhost:8080/admin/realms/myreaml/identity-provider/instances/google)
I got this:

```
{
   "alias":"google",
   "internalId":"8072227c-1876-4c63-abf0-f5ef951571b8",
   "providerId":"google",
   "enabled":true,
   "updateProfileFirstLoginMode":"on",
   "trustEmail":true,
   "storeToken":false,
   "addReadTokenRoleOnCreate":false,
   "authenticateByDefault":false,
   "linkOnly":false,
   "firstBrokerLoginFlowAlias":"first broker login",
   "config":{
      "hideOnLoginPage":"false",
      "acceptsPromptNoneForwardFromClient":"false",
      "clientId":"660041389399-tkvsvhf911dep04fs3fbbbeg5hngarbb.apps.googleusercontent.com",
      "disableUserInfo":"true",
      "syncMode":"import",
      "clientSecret":"**********"
   }
}
```

**Comment by rmartinc** [comment]
The `syncMode` should be in capitals `IMPORT`. Try just doing an update using the kcadm:

```
./kcadm.sh update identity-provider/instances/google -r myrealm -s "config.syncMode=IMPORT"
```

How did you create the provider? Using the console? The new admin console? kcadm?

**Comment by jderuere** [comment]
I used the new admin theme:
KC_FEATURES: admin2

**Comment by rmartinc** [comment]
Yep! I think this is an issue in the admin2 console. The default value is not doing the uppercase. To fix this just go the google provider, go to the end of teh page and in the `Sync mode` field just change to any other value and then again to import. Save. Check it was saved in capitals. Take care because in my case the field `Sync Mode` was hidden by the Save/Revert buttons (I needed to reduce the zoom in my browser).

**Comment by rmartinc** [comment]
I have created the bug in the ui project. This is not a server issue. Please @jderuere check that in capitals it works for you.

**Comment by jderuere** [comment]
I went into the DB and changed import to IMPORT and it worked.

**Comment by mposolda** [comment]
@rmartinc @jderuere Thanks for the investigation and confirm that IMPORT works.

This is fixed in Keycloak 18.0.2 according to underlying UI issue. So closing this.
